src/bikeracks/votes.py

```diff
@@ -51,120 +47,97 @@ def get_vote_status():
-            
-            query = "INSERT INTO votes (rack_id, user_id, vote_type) VALUES (?, ?, ?)"
-            db.execute(query, (rack_id, user_id, vote_type))
-            db.commit()
-            
-        # increment the upvote or downvote count
+        query = "UPDATE votes SET vote_type = ? WHERE rack_id = ? AND user_id = ?"
+        db.execute(query, (vote_type, rack_id, user_id))
+        db.commit()
+        # also update the vote count in bikeracks table
         if vote_type == 1:
-            # update the upvote_count in bikeracks for this rack, increment by 1
-            h.increment_upvote_count(rack_id, db)
+            # update the downvote_count in bikeracks for this rack, decrement by 1
+            h.decrement_downvote_count(rack_id, db)
+            
+        # if the user downvoted
         elif vote_type == -1:
-            # update the downvote_count in bikeracks for this rack, increment by 1
-            h.increment_downvote_count(rack_id, db)
+            # update the upvote_count in bikeracks for this rack, decrement by 1
+            h.decrement_upvote_count(rack_id, db)
         
-        resp = get_vote_data(rack_id, user_id)
-        return jsonify(resp)
-    except sqlite3.Error as e:
-        print("Database error:", e)
-        return "", 500
-    except KeyError as key_e:
-        print("Error with key: {}".format(key_e))
-        return "", 500
-    # TODO do I actually need below line
-   # maybe return 500 here
-    return "", 500
+    else:
+        # voting for the first time
+        
+        query = "INSERT INTO votes (rack_id, user_id, vote_type) VALUES (?, ?, ?)"
+        db.execute(query, (rack_id, user_id, vote_type))
+        db.commit()
+        
+    # increment the upvote or downvote count
+    if vote_type == 1:
+        # update the upvote_count in bikeracks for this rack, increment by 1
+        h.increment_upvote_count(rack_id, db)
+    elif vote_type == -1:
+        # update the downvote_count in bikeracks for this rack, increment by 1
+        h.increment_downvote_count(rack_id, db)
+    
+    resp = get_vote_data(rack_id, user_id)
+    return jsonify(resp)
     
 @votes.route('/unvote', methods=['POST'])
 def unvote():
-    if request.method != 'POST':
-        return "Bad Request", 400
     # remove a vote that the user had previously made
-    try:
-        rack_id = request.args.get('rack_id') or None
-        user_id = request.args.get('user_id') or None
-        vote_type = request.args.get('vote_type') or None
-        
-        # connect to db
-        db = get_db()
-        
-        # remove user's vote for this rack from votes table and update bikeracks vote count
-        
-        #remove user's vote 
-        query = """
-                DELETE FROM
-                    votes
+
+    rack_id = request.args.get('rack_id', None, type=int)
+    user_id = request.args.get('user_id', "", type=str)
+    vote_type = request.args.get('vote_type', None, type=int)
```

Comment:
None of these default values match the `type`

Reply:
Dropped the default types.